File: SOLID/DependencyInversion/BookingTest.cs
using Moq;
using SOLID.DependencyInversion.domain;
using SOLID.DependencyInversion.domain.booking;
using Xunit;

namespace SOLID.DependencyInversion
{
    public class BookingTest
    {
        private Mock<IAvailabilityDao> availabilities;
        private BookingService booking;

        public BookingTest()
        {
            availabilities = new Mock<IAvailabilityDao>();
            booking = new BookingService(availabilities.Object);
        }

        [Fact]
        public void should_book_when_available()
        {
            availabilities.Setup(a => a.IsAvailable()).Returns(true);
            BookingOutcome outcome = booking.Book();

            Assert.True(outcome.IsSuccessful());
        }

        [Fact]
        public void should_not_book_when_not_available()
        {
            availabilities.Setup(a => a.IsAvailable()).Returns(false);

            BookingOutcome outcome = booking.Book();

            Assert.False(outcome.IsSuccessful());
        }
    }
}

File: Java/Exercices/workspaces/solid-dip-end/src/test/java/zenika/solid/dip/domain/booking/BookingTest.java
package zenika.solid.dip.domain.booking;

import org.junit.Before;
import org.junit.Test;
import zenika.solid.dip.domain.availability.Availabilities;

import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class BookingTest {
    
    private Availabilities availabilities;
    private BookingService booking;
    
    @Before public void setup() {
        availabilities = mock(Availabilities.class);
        booking = new BookingService(availabilities);
    }  
    
    @Test public void should_book_when_available() {
        when(availabilities.isAvailable()).thenReturn(true);
        
        BookingOutcome outcome = booking.book();
        
        assertTrue(outcome.isSuccessful());
    } 
    
    @Test public void should_not_book_when_not_available() {
        when(availabilities.isAvailable()).thenReturn(false);
        
        BookingOutcome outcome = booking.book();
        
        assertFalse(outcome.isSuccessful());
    }
    
}
